src/device.hpp

```diff
@@ -46,10 +56,10 @@ namespace Deltacast
         void enable_loopback(int index);
         void disable_loopback(int index);
         bool wait_for_incoming_signal(int rx_index, const std::atomic_bool& stop_is_requested);
-        SignalInformation get_incoming_signal_information(int rx_index);
 
-        bool wait_genlock_locked(const std::atomic_bool& stop_is_requested);
-        bool configure_genlock(int genlock_source_rx_index, SignalInformation signal_info);
+        bool wait_sync_locked(const std::atomic_bool& stop_is_requested, std::unique_ptr<Helper::VideoInformation>& video_info);
+        bool configure_sync(int genlock_source_rx_index, std::unique_ptr<Helper::VideoInformation>& video_info);
+        std::unique_ptr<Helper::VideoInformation> get_video_information_for_channel(int index, Direction direction);
```

Comment:
As it stands for a factory method, I would suggest a different name to reflect that it creates an instance of a video_information based on the underlying technology of the channel.

Reply:
Something like : factory_create_video_information_for_channel ?